### PR TITLE
azure_cloud_setting

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

File: sdk/core/azure-core/azure/core/settings.py [hunk 1]
```diff
@@ -114,6 +115,26 @@ def convert_logging(value: Union[str, int]) -> int:
...
+
+    """
+    if isinstance(value, str):
+        azure_clouds = {cloud.name: cloud for cloud in AzureClouds}
+        if value in azure_clouds:
+            return azure_clouds[value]
+        raise ValueError(
+            "Cannot convert {} to Azure Cloud, valid values are: {}".format(value, ", ".join(azure_clouds.keys()))
+        )
+    return value
```
[the remark this answers is not in the document]
Comment (xiangyan99) [reply sdk/core/azure-core/azure/core/settings.py:135]: If it is not a string, it is a AzureClouds object. We don't need to convert it.

File: sdk/core/azure-core/azure/core/_azure_clouds.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+# pylint: disable=enum-must-inherit-case-insensitive-enum-meta
+
+from enum import Enum
+
+
+class AzureClouds(str, Enum):
```
[the remark this answers is not in the document]
Comment (xiangyan99) [reply sdk/core/azure-core/azure/core/_azure_clouds.py:31]: We don't want user to input a case insensitive string as the value of AzureClouds.

User should always use the enum.

**Review by annatisch — APPROVED** [review]

File: sdk/core/azure-core/azure/core/settings.py [hunk 3]
```diff
@@ -114,6 +115,28 @@ def convert_logging(value: Union[str, int]) -> int:
     return level
 
 
+def convert_azure_cloud(value: Union[str, AzureClouds]) -> AzureClouds:
```
Comment (johanste) [inline sdk/core/azure-core/azure/core/settings.py:118]: Do we need this to be public?

Comment (xiangyan99) [reply sdk/core/azure-core/azure/core/settings.py:118]: Other converters are all public. I don't think we need to treat it specially.

**Review by johanste — APPROVED** [review]

**Pull request merged**